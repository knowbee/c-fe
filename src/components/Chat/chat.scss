.chat {
  overflow-x: hidden;

  .profile-picture{
    border-radius: 68%;
    margin-right: 20px;
  }
  .header {
    position: fixed;
    width: 100vw;
    border-bottom: 1px solid #ddd;
    display: flex;
    justify-content: space-between;

    .dropdown-button {
      position: fixed;
      right: 1rem;
      top: 0.8rem;
      cursor: pointer;
      background-color: #fff;
      border: 1px solid #ddd;
      display: none;
    }
  }
  .chat-body {
    display: flex;
  }
}

@media (max-width: 999px) {
  .header {
    .dropdown-button {
      display: block !important;
      .sidebar {
        position: fixed;
        left: 1rem;
        background-color: white;
        box-shadow: 10px 10px 50px #888888;
        z-index: 99;
      }
    }
  }

  .chat-body {
    .sidebar {
      display: none;
    }

    .message {
      width: 100% !important;
      margin-left: 0rem;
    }
  }
}
