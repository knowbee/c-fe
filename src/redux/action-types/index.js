export * from "./registerUserTypes";
export * from "./loginTypes";
export * from "./currentUserTypes";
export * from "./usersTypes";
