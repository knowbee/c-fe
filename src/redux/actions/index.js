export * from "./registerActions";
export * from "./loginActions";
export * from "./currentUserActions";
