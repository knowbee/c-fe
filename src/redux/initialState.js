module.exports = {
  currentUser: {
    message: "",
    isLoggedIn: localStorage.getItem("token") !== null,
    profile: {},
  },
  register: {
    submitting: false,
    message: "",
    email: "",
    name: "",
    password: "",
    errors: {},
  },
  users: {
    fetching: false,
    chats: [],
    errors: {},
  },
  login: {
    submitting: false,
    credentials: {
      email: "",
      password: "",
    },
  },
};
