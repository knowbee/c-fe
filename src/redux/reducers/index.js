import { combineReducers } from "redux";
import register from "./registerReducer";
import login from "./loginReducer";
import currentUser from "./currentUserReducer";
import users from "./usersReducer";

const reducer = combineReducers({
  register,
  login,
  currentUser,
  users,
});
export default reducer;
