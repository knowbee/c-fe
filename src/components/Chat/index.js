import React, { useEffect } from "react";
import "./chat.scss";
import { connect } from "react-redux";
import Message from "../Message/index";
import ConversationsList from "../ConversationsList/index";
import { fetchUsers } from "../../redux/actions";
import menu from "./menuIcon.svg";

function Chat({ fetchUsers, chats, receiver_name }) {
  const [showSidebar, setShowSidebar] = React.useState(false);
  const Toggle = () => setShowSidebar(true);
  const token = localStorage.getItem("token");

  useEffect(() => {
    if (token) {
      fetchUsers().then((res) => {});
    }
  }, [fetchUsers, token]);

  const logout = () => {
    localStorage.clear();
    window.location.reload();
  };
  return (
    <div className="chat">
      <div className="header">
        <img
          style={{ width: "auto" }}
          src="https://iili.io/fi0ROQ.png"
          alt=""
        />
        <h4>{receiver_name}</h4>
        <button className="logout-button" onClick={logout}>
          Logout
        </button>
        <button
          className="dropdown-button"
          onClick={() => {
            Toggle();
          }}
        >
          <img src={menu} alt="menu" />
          {showSidebar ? <ConversationsList chats={chats} /> : null}
        </button>
      </div>
      <div className="chat-body">
        <ConversationsList chats={chats} />
        <Message />
      </div>
    </div>
  );
}

const mapStateToProps = ({
  users: { fetching, chats, errors },
  chat: { receiver_name },
}) => ({
  fetching,
  receiver_name,
  chats,
  errors,
});

export const mapDispatchToProps = (dispatch) => ({
  fetchUsers: () => dispatch(fetchUsers()),
});
export default connect(mapStateToProps, mapDispatchToProps)(Chat);
