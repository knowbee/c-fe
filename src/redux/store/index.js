import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension/developmentOnly";
import initialState from "../initialState";
import reducers from "../reducers/index";

const store = createStore(
  reducers,
  initialState,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
