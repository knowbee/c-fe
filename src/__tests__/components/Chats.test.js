import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore } from "redux";

import "./index.css";
import rootReducer from "../../redux/reducers";
import Chat from "../../components/Chat";

const store = createStore(rootReducer);

test("Chat page loads with default state", () => {
  const component = renderer.create(
    <Provider store={store}>
      <Chat />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
