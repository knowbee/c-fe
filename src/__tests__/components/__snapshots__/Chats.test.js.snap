// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat page loads with default state 1`] = `
<div
  className="chat"
>
  <div
    className="header"
  >
    <img
      alt=""
      src="https://iili.io/fi0ROQ.png"
      style={
        Object {
          "width": "auto",
        }
      }
    />
    <h4>
      
    </h4>
    <button
      className="logout-button"
      onClick={[Function]}
    >
      Logout
    </button>
    <button
      className="dropdown-button"
      onClick={[Function]}
    >
      <img
        alt="menu"
        src="test-file-stub"
      />
    </button>
  </div>
  <div
    className="chat-body"
  >
    <div
      className="sidebar"
    >
      <div
        className="sidebar-header"
      >
         
        <div
          className="active"
        >
           
          <p>
            Messages
          </p>
        </div>
      </div>
      <div
        className="sidebar-body"
      >
        <ul />
      </div>
      <div />
    </div>
    <div
      className="message"
    >
      <div
        className="chats"
      >
        <h2
          className="no-chat"
        >
          No chat selected
        </h2>
      </div>
    </div>
  </div>
</div>
`;
